### PR TITLE
minor: Use single Dockerfile with multiple build targets

### PR DESCRIPTION
**Pull request by patgrudniewski**
When adding new PHP version support ([example](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/6833)) it's required to copy&paste `Dockerfile` and `xdebug.ini` even if all of them are exactly (almost, only difference is PHP version number) the same to each other.

It can be solved with simple argument, so new version support is defined at `docker-compose.yml` only.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,6 @@
-FROM php:7.4-cli-alpine3.16
+ARG PHP_VERSION
```
Comment (Wirone) [inline Dockerfile:1]:
```suggestion
ARG PHP_VERSION
ARG XDEBUG_VERSION=3.1.2
```

And then `xdebug-${XDEBUG_VERSION}` below. In #6833 I used 3.2.0 where support for PHP 7.4 was dropped and we need differentiate XDebug version between builds.

Maybe 3.2.0 can be used as default and then 3.1.2 explicitly provided as arg for 7.4 build.

**Review by Wirone — CHANGES_REQUESTED** [review]
Please add `pcntl` extension too, like it was added in #6833. It's required for PHPUnit executed within container - without it there's an error message:

> Error:         PHP extension pcntl is required for enforcing time limits

PS. I wanted to clean it up after #6833 gets merged (it was added by convention as I saw it as higher priority to provide 8.2 runtime since we released official support for it). But great to see your contribution 👍 🙂

**Comment by Wirone** [comment]
@patgrudniewski your PR requires rebase, because 8.2 support was merged to main branch and it needs to be added to your unified `Dockerfile` too 🙂.

**Comment by patgrudniewski** [comment]
> PS. I wanted to clean it up after #6833 gets merged (it was added by convention as I saw it as higher priority to provide 8.2 runtime since we released official support for it). But great to see your contribution 👍 🙂

There are few more things to do with docker stuff here :) Unfortunately I had not enough time last few weeks, but I'm planning to do it more often

**Comment by Wirone** [comment]
Thank you @patgrudniewski and welcome within the contributors 🍻🎉

**Pull request merged**